src/Nethermind/Nethermind.JsonRpc/Modules/Admin/AdminModule.cs

```diff
@@ -17,40 +17,54 @@
 using System;
 using System.Linq;
 using System.Threading.Tasks;
-using Nethermind.Logging;
+using Nethermind.Core;
 using Nethermind.Network;
 
 namespace Nethermind.JsonRpc.Modules.Admin
 {
     public class AdminModule : IAdminModule
     {
-        private readonly ILogger _logger;
         private readonly IPeerManager _peerManager;
         private readonly IStaticNodesManager _staticNodesManager;
 
-        public AdminModule(ILogManager logManager, IPeerManager peerManager, IStaticNodesManager staticNodesManager)
+        public AdminModule(IPeerManager peerManager, IStaticNodesManager staticNodesManager)
         {
-            _logger = logManager.GetClassLogger();
             _peerManager = peerManager ?? throw new ArgumentNullException(nameof(peerManager));
             _staticNodesManager = staticNodesManager ?? throw new ArgumentNullException(nameof(staticNodesManager));
         }
         
-        public async Task<ResultWrapper<string>> admin_addPeer(string enode)
+        public async Task<ResultWrapper<string>> admin_addPeer(string enode, bool addToStaticNodes = false)
         {
-            var added = await _staticNodesManager.AddAsync(enode);
+            bool added;
+            if (addToStaticNodes)
+            {
+                added = await _staticNodesManager.AddAsync(enode);
+            }
+            else
+            {
+                added = _peerManager.AddPeer(new NetworkNode(enode));
+            }
 
             return added
                 ? ResultWrapper<string>.Success(enode)
-                : ResultWrapper<string>.Fail("Static node already exists.");
+                : ResultWrapper<string>.Fail("Failed to add peer.");
         }
 
-        public async Task<ResultWrapper<string>> admin_removePeer(string enode)
+        public async Task<ResultWrapper<string>> admin_removePeer(string enode, bool removeFromStaticNodes = false)
         {
-            var removed = await _staticNodesManager.RemoveAsync(enode);
-
+            bool removed;
+            if (removeFromStaticNodes)
+            {
+                removed = await _staticNodesManager.RemoveAsync(enode);
```

Comment:
does it also disconnect?

Reply:
ok, I see the tests

Reply:
since we split this functionality, should we keep it separate then instead of maintaining the connection via event + EventHandler?
then it would be somthing like:

            if (removeFromStaticNodes)
            {
                await _staticNodesManager.AddAsync(enode);
            }

            removed = _peerManager.RemovePeer(new NetworkNode(enode));

Reply:
I would keep it as is as static nodes manager might be accessed in the future in a different way from other part of the app (or for example it can listen to file modifications)